File: 백준/Silver/1743. 음식물 피하기/음식물 피하기.py
import sys
input = sys.stdin.readline
from collections import deque

def bfs(i,j):
    queue.append([i,j])
    visited[i][j] = True
    size = 1

    while queue:
        x,y = queue.popleft()
        for k in range(4):
            nx = x+dx[k]
            ny = y+dy[k]
            if 0<=nx<n and 0<=ny<m and not visited[nx][ny]:
                if board[nx][ny] == '#':
                    size += 1
                    queue.append([nx,ny])
                    visited[nx][ny] = True
                else:
                    continue
    
    return size

n,m,k = map(int,input().split())
answer = []
board = [['.']*m for _ in range(n)]
visited = [[False]*m for _ in range(n)]

for _ in range(k):
    a,b = map(int,input().split())
    board[a-1][b-1] = '#'

queue = deque([])
dx = [-1,1,0,0]
dy = [0,0,-1,1]

for i in range(n):
    for j in range(m):
        if board[i][j] == '#':
            cnt = bfs(i,j)
            answer.append(cnt)

print(max(answer))

File: 백준/Silver/1743. 음식물 피하기/음식물 피하기.js
const fs = require('fs');
/**정답 제출 시에는 이 코드로 제출 */
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
let answer = [];

function bfs(i,j){
    queue.push([i,j]);
    visited[i][j] = true;
    let size = 1; // 음식물 크기 

    while(queue.length){
        let [x,y] = queue.shift();
        for(let k=0; k<4; k++){
            let nx = x+dx[k]
            let ny = y+dy[k]
            if(0<=nx && nx<n && 0<=ny && ny<m && !visited[nx][ny]){
                if(board[nx][ny] === '#'){
                    size += 1
                    queue.push([nx,ny])
                    visited[nx][ny] = true
                } else {
                    continue
                }
            }
        }
    }

    return size;
}

const [n,m,_] = input.shift().split(' ').map(Number);
let board = Array.from(Array(n),() => Array(m).fill('.'));
let visited = Array.from(Array(n),() => Array(m).fill(false));

input.forEach((v) => {
    let [vx,vy] = v.split(' ').map(Number);
    // 음식물 쓰레기(#) 배치하기
    board[vx-1][vy-1] = '#';
});

let dx = [-1,1,0,0];
let dy = [0,0,-1,1];
let queue = [];

for(let r=0; r<n; r++){
    for(let c=0; c<m; c++){
        if(board[r][c] === '#' && !visited[r][c]){
            let cnt = bfs(r,c);
            answer.push(cnt)
        }
    }
}

console.log(Math.max(...answer));